src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const group: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    if (!group[`${item[key]}`]) {
+      group[`${item[key]}`] = [];
+    }
+
+    group[`${item[key]}`].push(item);
+  });
```

Comment:
```suggestion
  const prop = `${item[key]}`;

  items.forEach((item) => {
    if (!group[prop]) {
      group[prop] = [];
    }

    group[prop].push(item);
  });
```
You use `${item[key]}` for 3 times, I guess you should create variable for it